### DISCUSSION TITLE
Css framework

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#10] [General]
What should be used as css framework? 

I would like to see https://tailwindcss.com/ , any other ideas/options?

**Comment by CarlSargunar** [comment]
Will leave for the front enders - industry standards rather than latest and greatest is my vote, whatever that ends up being

**Comment by danbramall** [comment]
A dependency on Tailwind shouldn't be taken lightly.  It would likely require a front-end built process if promoting 'good practice' (e.g. to purge unused rules) which may not be necessary otherwise.  It's also not a trivial thing to extract from a project.  There is a not insignificant learning curve too, not so much in the utility-first concept but in learning the conventions and knowing when to abstract components etc.  I hope this isn't too negative :)  I'm not saying "don't do it", more "know what you're getting into".

**Reply by TimGeyssens** [reply]
sure valid points, that's why we have the Discussion :) I'm don't do a lot of css so not sure how big of a change it is compared to bootstrap or others like that

**Reply by danbramall** [reply]
It's very different to the traditional CSS frameworks like Bootstrap.  It tends to divide opinion as 100% utility-first is quite an extreme approach to CSS, still.  I don't know what the real-world adoption is but my feeling would be that it's still relatively low, so that might be a consideration if this project needs to have widespread appeal.

**Reply by deanleigh** [reply]
As this is essentially RAD I would recommend Bootstrap so you get the benefit of their JS as well as CSS and SCSS Partials.
I would be happy to help with this.

**Reply by TimGeyssens** [reply]
Sweet @deanleigh ! If v5 is an option then yes! https://v5.getbootstrap.com/ but don't want the extra jquery dependency in older versions

**Reply by deanleigh** [reply]
I started converting the current starter kit to Bootstrap 4 but concluded it was not worth the effort as so much has changed.
https://umbraco87bootstrap4starter20200825082902.azurewebsites.net/
Happy to use Bootstrap 5

**Reply by TimGeyssens** [reply]
@deanleigh sweet! we'll setup some tasks in different expertises before October...

**Reply by louisjrdev** [reply]
IMO using bootstrap has as much if not more overhead than using Tailwind, and ultimately are both utility styled frameworks.
👍 Tailwind haha

**Reply by deanleigh** [reply]
What would you use to replace the of-the-shelf JS Bootstrap provides ?
Unless I am mistaken Tailwind only give you the CSS?

**Reply by louisjrdev** [reply]
I would say vanilla for anything like that, or if you are referring to their js-based components like tabs etc, I would add a sprinkle of alpine.js as required.

_Edit for clarity_

**Reply by deanleigh** [reply]
Well their top nav makes life very easy for example

**Reply by louisjrdev** [reply]
Also bootstrap is a bit more bloated, see here:
Bootstrap: 61.7kB [Bundlephobia](https://bundlephobia.com/result?p=bootstrap@5.0.0-alpha1)

Tailwind: 1.8kB [Bundlephobia](https://bundlephobia.com/result?p=tailwindcss@1.8.10)
AlpineJS: 25.1kB [Bundlephobia](https://bundlephobia.com/result?p=alpinejs@2.7.0)
(Total 26.7kB) Minified

**Reply by deanleigh** [reply]
Fair, but it's scss partials so you only compile what you need

**Reply by louisjrdev** [reply]
Same with tailwind ;)

**Reply by TimGeyssens** [reply]
wow big difference :)

**Reply by madsoulswe** [reply]
Love Tailwind! 

It should be easy the configure. 

My only input:
- Tailwind config setup is more complex than Bootstrap scss variables.
- Bootstrap still rely on libsass, so dotnet compilation is easy.
- Complete Tailwind without purge is 2380.4kB uncompressed.
- Tailwind has ALOT more flexibility in the utilities, demands less custom styling.
- Bootstrap without components has a very small KB footprint.

=)

**Reply by deanleigh** [reply]
I would be happy to use either Bootstrap or Tailwind, I have a little experience with both.
If the purpose of this 'kit' to make getting stated with Umbraco as easy as possible for the most people I still think Bootstrap is a better option.
As front-ender I personally don't use any CSS frameworks but Umbraco seems a 'developer' led community and I know most devs prefer frameworks to writing CSS so happy to go with the consensus.

**Reply by louisjrdev** [reply]
I would disagree that the setup is much more complicated than a bootstrap one. 
I don't think it's worth mentioning tailwind _without_ purge as you would definitely purge it...
Tailwind also doesn't rely on sass to use only what you need, and I would argue is quicker to style things with.

Also everyone and their dog has made a bootstrap theme / starter on github, why not look to use something more modern and something that has a smaller learning curve from new.

Also I dont think setup / config is a big deal either way, as this project will cover that.

**Reply by deanleigh** [reply]
I am starting to wonder if @louisjrdev is really Adam Wathan ;-)

**Reply by madsoulswe** [reply]
I wrote a long reponse... but when reading through the discussions, this is not a startkit. 🤷🏻‍♂️
It's a Starter Template similar to https://github.com/umbraco/Package.Templates/tree/master/ProjectTemplates/UmbracoSite.1

Then I'm voting for Tailwind!

**Reply by louisjrdev** [reply]
@deanleigh :eyes: hmmm noo...
Just passionate about seeing umbraco adopted in areas it has not been previously and I think it's a great opportunity. It is much harder to have influence on these decisions with HQ, so if not HQ, then here! :smile:
